connector/signaltometricsconnector/config/config.go

```diff
@@ -0,0 +1,318 @@
+// based on their durations in histograms, summaries, or as 2 sum metrics.
+//
+// For all other event types, the connector can count the number of events.
+//
+// All metrics produced by the connector are in delta temporality.
+type Config struct {
+	Spans      []MetricInfo `mapstructure:"spans"`
+	Datapoints []MetricInfo `mapstructure:"datapoints"`
+	Logs       []MetricInfo `mapstructure:"logs"`
+}
+
+var _ confmap.Unmarshaler = (*Config)(nil)
+
+func (c *Config) Validate() error {
+	if len(c.Spans) == 0 && len(c.Datapoints) == 0 && len(c.Logs) == 0 {
+		return fmt.Errorf("no configuration provided, at least one should be specified")
+	}
+	var multiError error // collect all errors at once
+	if len(c.Spans) > 0 {
+		parser, err := ottlspan.NewParser(
+			ottlget.CustomFuncs[ottlspan.TransformContext](),
+			component.TelemetrySettings{Logger: zap.NewNop()},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to create parser for OTTL spans: %w", err)
+		}
+		for _, span := range c.Spans {
+			if err := validateMetricInfo(span, parser); err != nil {
+				multiError = errors.Join(multiError, fmt.Errorf("failed to validate spans configuration: %w", err))
+			}
+		}
+	}
+	if len(c.Datapoints) > 0 {
+		parser, err := ottldatapoint.NewParser(
+			ottlget.CustomFuncs[ottldatapoint.TransformContext](),
+			component.TelemetrySettings{Logger: zap.NewNop()},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to create parser for OTTL datapoints: %w", err)
+		}
+		for _, dp := range c.Datapoints {
+			if err := validateMetricInfo(dp, parser); err != nil {
+				multiError = errors.Join(multiError, fmt.Errorf("failed to validate datapoints configuration: %w", err))
+			}
+		}
+	}
+	if len(c.Logs) > 0 {
+		parser, err := ottllog.NewParser(
+			ottlget.CustomFuncs[ottllog.TransformContext](),
+			component.TelemetrySettings{Logger: zap.NewNop()},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to create parser for OTTL logs: %w", err)
+		}
+		for _, log := range c.Logs {
+			if err := validateMetricInfo(log, parser); err != nil {
+				multiError = errors.Join(multiError, fmt.Errorf("failed to validate logs configuration: %w", err))
+			}
+		}
+	}
+	return multiError
+}
+
+// Unmarshal with custom logic to set default values.
+// This is necessary to ensure that default metrics are
+// not configured if the user has specified any custom metrics.
+func (c *Config) Unmarshal(componentParser *confmap.Conf) error {
+	if componentParser == nil {
+		// Nothing to do if there is no config given.
+		return nil
+	}
+	if err := componentParser.Unmarshal(c, confmap.WithIgnoreUnused()); err != nil {
+		return err
+	}
+	for i, info := range c.Spans {
+		info.ensureDefaults()
+		c.Spans[i] = info
+	}
+	for i, info := range c.Datapoints {
+		info.ensureDefaults()
+		c.Datapoints[i] = info
+	}
+	for i, info := range c.Logs {
+		info.ensureDefaults()
+		c.Logs[i] = info
+	}
+	return nil
+}
+
+// MetricInfo for a data type
+type MetricInfo struct {
```

Comment:
I use `MetricDef` for the parsed config (in the `model/*`), maybe we can use `MetricConfig` here? WDYT?

Reply:
Does it matter if they have the same name? I think it's still a metric definition from the user's perspective. I wouldn't go with MetricConfig, since it's in the config package.

Anyway, not critical that we address this atm, we can come back to it.